src/mcpadapt/smolagents_adapter.py

```diff
@@ -104,12 +107,35 @@ def forward(self, *args, **kwargs) -> str:
                         f"tool {self.name} returned multiple content, using the first one"
                     )
 
-                if not isinstance(mcp_output.content[0], mcp.types.TextContent):
-                    raise ValueError(
-                        f"tool {self.name} returned a non-text content: `{type(mcp_output.content[0])}`"
-                    )
+                content = mcp_output.content[0]
+
+                if isinstance(content, mcp.types.TextContent):
+                    return content.text
+
+                if isinstance(content, mcp.types.ImageContent):
+                    from PIL import Image
+
+                    image_data = base64.b64decode(content.data)
+                    image = Image.open(BytesIO(image_data))
+                    return image
+
+                if isinstance(content, mcp.types.AudioContent):
+                    if not _is_package_available("torchaudio"):
+                        raise ValueError(
+                            "Audio content requires the torchaudio package to be installed. "
+                            "Please install it with `pip install torchaudio`.",
```

Comment:
here maybe we should recommend to use the extra dependecy group like:
`uv add mcpadapt[smolagents,audio]` instead of isntalling torchaudio separetly?